core/test/element_arithmetic_test.cpp

```diff
@@ -43,3 +43,35 @@ TEST_F(ElementArithmeticTest, non_in_place) {
 }
 
 TEST_F(ElementArithmeticTest, unary_minus) { EXPECT_EQ(unary_minus(a), -a); }
+
+TEST(ElementArithmeticIntegerDivisionTest, truediv) {
+  // 32 bit ints gives double not float, consistent with numpy
+  EXPECT_TRUE((std::is_same_v<decltype(divide(int32_t{}, int32_t{})), double>));
+  EXPECT_TRUE((std::is_same_v<decltype(divide(int64_t{}, int64_t{})), double>));
+}
+
+TEST(ElementArithmeticIntegerDivisionTest, truediv_32bit) {
+  const int32_t a = 2;
```

Comment:
I'm slightly confused as to why we are checking division of integer by integer, if we state above that
```
we do *not* support any integer type as left-hand-side, to match Python 3 and numpy "truediv" behavior.
```

Reply:
ok, i think i misunderstood. The int on the left had side is only for in-place, correct?

Reply:
Correct. 👍